InformationScripting/src/queries/AstQuery.cpp

```diff
@@ -264,12 +300,18 @@ void AstQuery::addCallInformation(TupleSet& ts, OOModel::Method* method, QList<O
 	}
 }
 
-void AstQuery::addNodesOfType(TupleSet& ts, const QString& typeName, Model::Node* from)
+void AstQuery::addNodesOfType(TupleSet& ts, const Model::SymbolMatcher& matcher, Model::Node* from)
 {
 	if (!from) from = target_->root();
-	auto allNodeOfType =  AllNodesOfType::allNodesOfType(from, typeName);
+	auto allNodeOfType =  AllNodesOfType::allNodesOfType(from, matcher);
 	for (auto node : allNodeOfType)
 		ts.add({{"ast", node}});
 }
 
+Model::SymbolMatcher AstQuery::matcherFor(const QString& text)
```

Comment:
I think at some point we should control the type of matcher based on command arguments. E.g. no argument -> simple substring matcher (*text*). Otherwise a fullblown regex matcher. But maybe some other smart option that detects the type of pattern is also fine. What do you think?